compiler/src/dmd/targetcompiler.d

```diff
@@ -0,0 +1,211 @@
+version (IN_GCC)
+{
+    mixin template VarDeclarationExtra() { }
+}
+else version (IN_LLVM)
+{
+    mixin template VarDeclarationExtra() { }
+}
+else version (MARS)
+{
+    mixin template VarDeclarationExtra()
+    {
+        bool inClosure;         /// is inserted into a GC allocated closure
+        bool inAlignSection;    /// is inserted into an aligned section on stack
+    }
+}
+else
+    static assert(0, "unknown compiler being built");
+
+/**********************************
+ * Extra Fields for FuncDeclaration
+ */
+version (IN_GCC)
+{
+    mixin template FuncDeclarationExtra() { }
+}
+else version (IN_LLVM)
+{
+    mixin template FuncDeclarationExtra() { }
+}
+else version (MARS)
+{
+    mixin template FuncDeclarationExtra()
+    {
+        VarDeclarations* alignSectionVars;  /// local variables with alignment needs larger than stackAlign
+	import dmd.backend.cc : Symbol;
+        Symbol* salignSection;              /// pointer to aligned section, if any
+    }
+}
+else
+    static assert(0, "unknown compiler being built");
+
+/**********************************
+ * Extra Fields for FuncDeclaration
+ */
+version (IN_GCC)
+{
+    mixin template alignSectionVarsExtra() { void doAlign() { } }
+}
+else version (IN_LLVM)
+{
+    mixin template alignSectionVarsExtra() { void doAlign() { } }
+}
+else version (MARS)
+{
+    /* If the alignment of a stack local is greater than the stack alignment,
+     * note it in the enclosing function's alignSectionVars
+     */
+    mixin template alignSectionVarsExtra()
+    {
+        void doAlign()
+        {
+            if (!dsym.alignment.isDefault() && sc.func &&
+                dsym.alignment.get() > target.stackAlign() &&
+                sc.func && !dsym.isDataseg() && !dsym.isParameter() && !dsym.isField())
+            {
+                auto fd = sc.func;
+                if (!fd.alignSectionVars)
+                    fd.alignSectionVars = new VarDeclarations();
+                fd.alignSectionVars.push(dsym);
+            }
+        }
+    }
+}
+else
+    static assert(0, "unknown compiler being built");
+
+
+/* Returns: true if v is in an align section
+ */
+mixin template alignSectionVarsContains()
+{
+    version (IN_GCC)
+    {
+        bool isAlignSectionVar(VarDeclaration v) { return false; }
+    }
+    else version (IN_LLVM)
+    {
+        bool isAlignSectionVar(VarDeclaration v) { return false; }
+    }
+    else version (MARS)
+    {
+        bool isAlignSectionVar(VarDeclaration v)
+        {
+            return fd.alignSectionVars && (*fd.alignSectionVars).contains(v);
+        }
+    }
+    else
+        static assert(0, "unknown compiler being built");
+}
+
+/* Returns: true if ansi color is to be used
+ */
+
+mixin template UseAnsiColors()
+{
+    bool useAnsiColors()
+    {
+        version (IN_GCC)
+        {
+            return false;
+        }
+        else version (IN_LLVM)
+        {
+            import dmd.console : detectTerminal;
+            return detectTerminal();
```

Comment:
LDC wants the DMD logic here, so incl. the `detectColorPreference()` (which looks at an env var).

Reply:
The original IN_LLVM does not call detectColorPreference().